cdi/src/test/java/org/hibernate/validator/test/cdi/internal/ValidateableBeanFilterTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.cdi.internal;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertFalse;
+
+import java.io.Serializable;
+import java.util.function.Predicate;
+
+import javax.validation.constraints.Min;
+import javax.validation.executable.ValidateOnExecution;
+
+import org.hibernate.validator.cdi.internal.ValidateableBeanFilter;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * @author Marko Bekhta
+ */
+public class ValidateableBeanFilterTest {
+
+	private Predicate<Class<?>> predicate;
+
+	@BeforeMethod
+	public void setUp() {
+		this.predicate = new ValidateableBeanFilter();
+	}
+
+	@Test
+	public void test() {
+		assertFalse( predicate.test( Object.class ) );
+		assertFalse( predicate.test( Foo.class ) );
+		assertFalse( predicate.test( Baz.class ) );
+		assertTrue( predicate.test( Bar.class ) );
+		assertTrue( predicate.test( FooBar.class ) );
+		assertTrue( predicate.test( BarBaz.class ) );
+	}
+
+	private static class Foo {
+
+		public void doNothing() {
+		}
+	}
+
+	private static class FooBar extends Foo implements Bar {
+
+		@Override public int bar() {
+			return 0;
+		}
+
+		public void doSomething(Foo foo) {
+		}
+	}
+
+	interface Bar {
+
+		@Min(10)
+		int bar();
+	}
+
+	private static class Baz implements Serializable, Predicate<Integer>, Comparable<Baz> {
+
+		@Override public int compareTo(Baz o) {
+			return 0;
+		}
+
+		@Override public boolean test(Integer integer) {
+			return false;
+		}
+	}
+
+	private static class BarBaz extends Baz implements Bar {
```

Comment:
I think, in this case, it would make sense to name the classes with the test case they're supposed to check. It's a bit hard to follow otherwise.